lib/install/management/delete.go

```diff
@@ -19,43 +19,80 @@ import (
 
 	log "github.com/Sirupsen/logrus"
 
+	"github.com/vmware/govmomi/vim25/types"
 	"github.com/vmware/vic/lib/metadata"
 	"github.com/vmware/vic/pkg/errors"
 	"github.com/vmware/vic/pkg/vsphere/compute"
+	"github.com/vmware/vic/pkg/vsphere/tasks"
 	"github.com/vmware/vic/pkg/vsphere/vm"
+
+	"golang.org/x/net/context"
 )
 
 func (d *Dispatcher) DeleteVCH(conf *metadata.VirtualContainerHostConfigSpec) error {
-	log.Infof("Removing VMs")
+	var errs []string
+
+	var err error
+	var vmm *vm.VirtualMachine
+
+	if vmm, err = d.findApplianceByID(conf); err != nil {
+		return err
+	}
+	if vmm == nil {
+		return nil
+	}
 
-	if err := d.DeleteVCHInstances(conf); err != nil {
+	if err = d.DeleteVCHInstances(vmm, conf); err != nil {
+		// if container delete failed, do not remove anything else
+		log.Infof("Specify --force to force delete")
 		return err
 	}
-	if err := d.destroyResourcePoolIfEmpty(conf); err != nil {
+
+	if err = d.DeleteStores(vmm, conf); err != nil {
+		errs = append(errs, err.Error())
+	}
+
+	if err = d.deleteNetworkDevices(vmm, conf); err != nil {
+		errs = append(errs, err.Error())
+	}
+	if err = d.removeNetwork(conf); err != nil {
+		errs = append(errs, err.Error())
+	}
+	if len(errs) > 0 {
+		// stop here, leave vch appliance there for next time delete
+		return errors.New(strings.Join(errs, "\n"))
+	}
+	err = d.deleteVM(vmm, true)
+	if err != nil {
+		log.Debugf("Error deleting appliance VM %s", err)
 		return err
 	}
-	return d.removeNetwork(conf)
+	if err = d.destroyResourcePoolIfEmpty(conf); err != nil {
```

Comment:
are you aware of the object.ResourcePool.DestroyChildren method in govmomi? I don't know how aggressive it is, but it may be worth trying here.